mvnxparser/mvnxStreamReader/mvnxStreamReader.h

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright: (C) 2017 iCub Facility
+ * Author: Diego Ferigo
+ * CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ *
+ */
+
+/**
+ * @file mvnxStreamReader.h
+ * @brief Validate and parse efficiently mvnx files
+ * @author Diego Ferigo
+ * @date 18/04/2017
+ */
+#ifndef MVNX_STREAM_READER_H
+#define MVNX_STREAM_READER_H
+
+#include "xmlDataContainers.h"
+#include "xmlStreamReader.h"
+#include <QXmlStreamReader>
+#include <iostream>
+#include <memory>
+#include <unordered_map>
+#include <vector>
+
+namespace xmlstream {
+namespace mvnx {
+
+// Simple container of the mvnx parsing configuration.
+typedef std::unordered_map<std::string, bool> mvnxConf;
+
+class mvnxStreamReader : public xmlstream::xmlStreamReader {
+private:
+    mvnxConf conf;
+    IContentPtrS xmlTreeRoot;
+    std::vector<IContentPtrS> elementsLIFO;
+
+public:
+    mvnxStreamReader() : xmlTreeRoot(nullptr){};
+    ~mvnxStreamReader() = default;
+
+    // Get methods
+    mvnxConf getConf() const { return conf; };
+    xmlContentPtrS getXmlTreeRoot() const
+    {
+        return std::dynamic_pointer_cast<xmlContent>(xmlTreeRoot);
+    };
```

Comment:
semicolon after method definition is not necessary.
(this comment added only once).
If you don't mind, please move the definition in the associated `.cpp` file